app/src/main/java/com/github/onlynotesswent/ui/overview/DeckOverview.kt

```diff
@@ -0,0 +1,129 @@
+package com.github.onlynotesswent.ui.overview
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.github.onlynotesswent.MainActivity
+import com.github.onlynotesswent.R
+import com.github.onlynotesswent.model.deck.DeckViewModel
+import com.github.onlynotesswent.model.folder.Folder
+import com.github.onlynotesswent.model.folder.FolderViewModel
+import com.github.onlynotesswent.model.user.UserViewModel
+import com.github.onlynotesswent.ui.common.EditDeckDialog
+import com.github.onlynotesswent.ui.common.FolderDialog
+import com.github.onlynotesswent.ui.navigation.BottomNavigationMenu
+import com.github.onlynotesswent.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.github.onlynotesswent.ui.navigation.NavigationActions
+import com.github.onlynotesswent.ui.navigation.Screen
+import com.google.firebase.Timestamp
+
+/**
+ * Displays the overview screen which contains a list of the user's decks and folders. The user can
+ * create new decks and folders from this screen. If there are no decks or folders, it shows a text
+ * to the user indicating that there are no decks or folders.
+ *
+ * @param navigationActions The navigationActions instance used to transition between different
+ *   screens.
+ * @param deckViewModel The ViewModel that provides the list of decks to display.
+ * @param userViewModel The ViewModel that provides the current user.
+ * @param folderViewModel The ViewModel that provides the list of folders to display.
+ */
+@Composable
+fun DeckOverviewScreen(
```

Comment:
Really nice Screen as you don't reinvent the wheel by remaking composable that are already defined generically elsewhere.